lib/datadog/tracing/distributed/helpers.rb

```diff
@@ -20,54 +20,27 @@ def self.clamp_sampling_priority(sampling_priority)
           sampling_priority
         end
 
-        def self.truncate_base16_number(value)
-          # Lowercase if we want to parse base16 e.g. 3E8 => 3e8
-          # DEV: Ruby will parse `3E8` just fine, but to test
-          #      `num.to_s(base) == value` we need to lowercase
-          value = value.downcase
+        def self.parse_decimal_id(value)
+          return unless value
 
-          # Truncate to trailing 16 characters if length is greater than 16
-          # https://github.com/apache/incubator-zipkin/blob/21fe362899fef5c593370466bc5707d3837070c2/zipkin/src/main/java/zipkin2/storage/StorageComponent.java#L49-L53
-          # DEV: This ensures we truncate B3 128-bit trace and span ids to 64-bit
-          value = value[value.length - 16, 16] if value.length > 16
-
-          # Remove any leading zeros
-          # DEV: When we call `num.to_s(16)` later Ruby will not add leading zeros
-          #      for us so we want to make sure the comparision will work as expected
-          # DEV: regex, remove all leading zeros up until we find the last 0 in the string
-          #      or we find the first non-zero, this allows `'0000' -> '0'` and `'00001' -> '1'`
-          value.sub(/^0*(?=(0$)|[^0])/, '')
-        end
-
-        def self.value_to_id(value, base: 10)
-          id = value_to_number(value, base: base)
-
-          # Return early if we could not parse a number
-          return if id.nil?
+          value = value.to_s
+          num   = value.to_i
 
-          # Zero or greater than max allowed value of 2**64
-          return if id.zero? || id > Tracing::Utils::EXTERNAL_MAX_ID
+          return unless num.to_s(10) == value
 
-          id < 0 ? id + (2**64) : id
+          num
         end
 
-        def self.value_to_number(value, base: 10)
-          # It's important to make a difference between no data and zero.
-          return if value.nil?
-
-          # Be sure we have a string
-          value = value.to_s
+        def self.parse_hex_id(value, length: nil)
+          return unless value
 
-          # If we are parsing base16 number then truncate to 64-bit
-          value = Helpers.truncate_base16_number(value) if base == 16
+          value = value.to_s.downcase
+          value = value[value.length - length, length] if length && value.length > length
+          value = value.sub(/^0*(?=(0$)|[^0])/, '')
```

Comment:
Let's add a comment here that we'd like to remove these extra operations on hex ids, now that we support 128-bit a lot of validation Just Works™️.